homeassistant/components/purpleair/config_flow.py

```diff
@@ -178,6 +180,38 @@ async def async_step_by_coordinates(
 
         return await self.async_step_choose_sensor()
 
+    async def async_step_check_api_key(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the step to check the API key."""
+        if self._reauth_api_key:
+            error_step_id = "reauth_confirm"
+        else:
+            error_step_id = "user"
+
+        api_key = self._flow_data[CONF_API_KEY]
+
+        validation = await async_validate_api_key(self.hass, api_key)
+
+        if validation.errors:
+            return self.async_show_form(
+                step_id=error_step_id,
+                data_schema=API_KEY_SCHEMA,
+                errors=validation.errors,
+            )
+
+        if self._reauth_api_key:
+            if existing_entry := await self.async_set_unique_id(self._reauth_api_key):
```

Comment:
Can we use the account id for the unique id instead of the api key?

Reply:
The PurpleAir API doesn't give any sort of account ID that I'm aware of—you just get an API key. Am I missing something?

Reply:
It doesn't look like there is a way in their API to get the account identifier for the api key 🤦 

https://api.purpleair.com/#api-welcome-using-api-keys

Reply:
Its a bit confusing that you change the api key but the unique id is still the old api key

I think it might be better to not have a unique id for the integration and add a check `self._async_abort_entries_match({...})` to prevent them from setting up multiple entries with the same api key